compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPULowerToUKernels.cpp

```diff
@@ -364,36 +365,34 @@ matchDAGForUKernel(RewriterBase &rewriter, IREE::Codegen::QueryTileSizesOp op) {
   if (tensorType.getRank() != 2) {
     return rewriter.notifyMatchFailure(op, "only the 2D case is implemented");
   }
-  auto encoding = getEncoding(tensorType);
+  auto encoding = tensorType.getEncoding()
+                      .dyn_cast_or_null<IREE::LinalgExt::EncodingAttr>();
   if (!encoding) {
     return rewriter.notifyMatchFailure(op, "no TensorEncoding attribute");
   }
-  auto matmulType = getMatmulType(*encoding);
-  auto matmulOperandRole = getMatmulOperandRole(*encoding);
-  if (!matmulType || !matmulOperandRole) {
-    return rewriter.notifyMatchFailure(op, "unhandled TensorEncoding");
-  }
+  auto opKind = encoding.getOpKind().getValue();
+  auto role = encoding.getRole().getValue();
   uint32_t flags = 0;
-  if (*matmulType == MatmulType::F32F32F32) {
+  if (opKind == IREE::LinalgExt::EncodingOpKind::MATMUL_F32F32F32) {
```

Comment:
is it better to use switch-case statements here?

Reply:
If that's OK, I'd like to keep the existing if-else chain. Besides the fact that it's how the existing code is structured, this is fewer lines of code: if this were a `switch`, each case would need a `break;`, which clang-format would likely put on its own separate line.  This should not make a difference either way in terms of the generated code, as LLVM is able to convert between if-else chains and switches.

Reply:
It's okay to use if-else chain. I think they are from table-gen and they are `enum`, not `enum class`. If they were `enum class` and we're using `switch-case` statements, the compiler can warn out missing cases. That's quite useful to avoid bugs in future changes. Since they are not `enum class`, I think it's okay to use `if-else` chain here.

Reply:
They are comming from table-gen and are actually `enum class` now. Maybe that changed recently. Hm, good point. OK, I'll make that change.